diff -ruN src.old/src/callbacks.cc src/src/callbacks.cc
--- src.old/src/callbacks.cc	2017-08-12 04:36:43.405189413 +0000
+++ src/src/callbacks.cc	2017-08-12 04:40:50.302774059 +0000
@@ -7255,8 +7255,11 @@
 		unit_inserted(u);
 	} else if(response == GTK_RESPONSE_HELP) {
 		GError *error = NULL;
-		//gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-unit-creation", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 		gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(unitedit_builder, "unit_edit_dialog")), "file://" PACKAGE_DOC_DIR "/html/qalculate-units.html#qalculate-unit-creation", gtk_get_current_event_time(), &error);
+#else
+		gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-unit-creation", gtk_get_current_event_time(), &error);
+#endif
 		if(error) {
 			gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 			GtkWidget *d = gtk_message_dialog_new (GTK_WINDOW(gtk_builder_get_object(unitedit_builder, "unit_edit_dialog")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help.\n%s"), error_str);			
@@ -7607,8 +7610,11 @@
 		function_inserted(f);
 	} else if(response == GTK_RESPONSE_HELP) {
 		GError *error = NULL;
-		//gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-function-creation", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 		gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(functionedit_builder, "function_edit_dialog")), "file://" PACKAGE_DOC_DIR "/html/qalculate-functions.html#qalculate-function-creation", gtk_get_current_event_time(), &error);
+#else
+		gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-function-creation", gtk_get_current_event_time(), &error);
+#endif
 		if(error) {
 			gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 			GtkWidget *d = gtk_message_dialog_new (GTK_WINDOW(gtk_builder_get_object(functionedit_builder, "function_edit_dialog")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help.\n%s"), error_str);
@@ -7792,8 +7798,11 @@
 		variable_inserted(v);
 	} else if(response == GTK_RESPONSE_HELP) {
 		GError *error = NULL;
-		//gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-variable-creation", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 		gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(unknownedit_builder, "unknown_edit_dialog")), "file://" PACKAGE_DOC_DIR "/html/qalculate-variables.html#qalculate-variable-creation", gtk_get_current_event_time(), &error);		
+#else
+		gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-variable-creation", gtk_get_current_event_time(), &error);
+#endif
 		if(error) {
 			gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 			GtkWidget *d = gtk_message_dialog_new (GTK_WINDOW(gtk_builder_get_object(unknownedit_builder, "unknown_edit_dialog")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help.\n%s"), error_str);
@@ -7966,8 +7975,11 @@
 		variable_inserted(v);
 	} else if(response == GTK_RESPONSE_HELP) {
 		GError *error = NULL;
-		//gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-variable-creation", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 		gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(variableedit_builder, "variable_edit_dialog")), "file://" PACKAGE_DOC_DIR "/html/qalculate-variables.html#qalculate-variable-creation", gtk_get_current_event_time(), &error);
+#else
+		gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-variable-creation", gtk_get_current_event_time(), &error);
+#endif
 		if(error) {
 			gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 			GtkWidget *d = gtk_message_dialog_new (GTK_WINDOW(gtk_builder_get_object(variableedit_builder, "variable_edit_dialog")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help.\n%s"), error_str);
@@ -8253,8 +8265,11 @@
 		variable_inserted(v);
 	} else if(response == GTK_RESPONSE_HELP) {
 		GError *error = NULL;
-		//gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-vectors-matrices", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 		gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(matrixedit_builder, "matrix_edit_dialog")), "file://" PACKAGE_DOC_DIR "/html/qalculate-variables.html#qalculate-vectors-matrices", gtk_get_current_event_time(), &error);
+#else		
+		gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-vectors-matrices", gtk_get_current_event_time(), &error);
+#endif
 		if(error) {
 			gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 			GtkWidget *d = gtk_message_dialog_new (GTK_WINDOW(gtk_builder_get_object(matrixedit_builder, "matrix_edit_dialog")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help.\n%s"), error_str);
@@ -14896,8 +14911,11 @@
 }
 void on_menu_item_help_activate(GtkMenuItem*, gpointer) {
 	GError *error = NULL;
-	//gtk_show_uri(NULL, "ghelp:qalculate-gtk", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 	gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(main_builder, "main_window")), "file://" PACKAGE_DOC_DIR "/html/index.html", gtk_get_current_event_time(), &error);
+#else
+	gtk_show_uri(NULL, "ghelp:qalculate-gtk", gtk_get_current_event_time(), &error);
+#endif
 	if(error) {
 		gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 		GtkWidget *dialog = gtk_message_dialog_new(GTK_WINDOW(gtk_builder_get_object(main_builder, "main_window")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help for Qalculate!.\n%s"), error_str);
@@ -15850,8 +15868,11 @@
 }
 void on_plot_button_help_clicked(GtkButton, gpointer) {
 	GError *error = NULL;
-	//gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-plotting", gtk_get_current_event_time(), &error);
+#if GTK_CHECK_VERSION(3, 22, 0)
 	gtk_show_uri_on_window(GTK_WINDOW(gtk_builder_get_object(plot_builder, "plot_dialog")), "file://" PACKAGE_DOC_DIR "/html/qalculate-plotting.html", gtk_get_current_event_time(), &error);	
+#else
+	gtk_show_uri(NULL, "ghelp:qalculate-gtk?qalculate-plotting", gtk_get_current_event_time(), &error);
+#endif
 	if(error) {
 		gchar *error_str = g_locale_to_utf8(error->message, -1, NULL, NULL, NULL);
 		GtkWidget *d = gtk_message_dialog_new (GTK_WINDOW(gtk_builder_get_object(plot_builder, "plot_dialog")), (GtkDialogFlags) 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE, _("Could not display help.\n%s"), error_str);
