diff --git a/libqalculate/Calculator.cc b/libqalculate/Calculator.cc
index a4fa85b..e27eef5 100644
--- a/libqalculate/Calculator.cc
+++ b/libqalculate/Calculator.cc
@@ -9194,7 +9194,7 @@ bool Calculator::fetchExchangeRates(int timeout) {
 	curl_global_cleanup();
 	if(res != CURLE_OK) {error(true, _("Failed to download exchange rates from ECB: %s"), error_buffer, NULL); return false;}
 	if(sbuffer.empty()) {error(true, _("Failed to download exchange rates from ECB: %s"), "Document empty", NULL); return false;}
-	ofstream file(getExchangeRatesFileName(), ios::out | ios::trunc | ios::binary);
+	ofstream file(getExchangeRatesFileName().c_str(), ios::out | ios::trunc | ios::binary);
 	if(!file.is_open()) {
 		error(true, _("Failed to download exchange rates from ECB: %s"), strerror(errno), NULL);
 		return false;
