diff --git a/libqalculate/util.cc b/libqalculate/util.cc
index f1be1b4..38b9b17 100644
--- a/libqalculate/util.cc
+++ b/libqalculate/util.cc
@@ -457,11 +457,7 @@ string getLocalDir() {
 	return str + "\\Qalculate";
 #	endif
 #else
-	const char *homedir;
-	if((homedir = getenv("XDG_CONFIG_HOME")) == NULL) {
-		return string(getpwuid(getuid())->pw_dir) + "/.config/qalculate";
-	}
-	return string(homedir) + "/qalculate";
+	return string(getenv("SNAP_USER_DATA"));
 #endif
 }
 string getLocalDataDir() {
@@ -480,11 +476,7 @@ string getLocalDataDir() {
 	return str + "\\Qalculate";
 #	endif
 #else
-	const char *homedir;
-	if((homedir = getenv("XDG_DATA_HOME")) == NULL) {
-		return string(getpwuid(getuid())->pw_dir) + "/.local/share/qalculate";
-	}
-	return string(homedir) + "/qalculate";
+	return string(getenv("SNAP_USER_COMMON"));
 #endif
 }
 string getLocalTmpDir() {
@@ -505,11 +497,7 @@ string getLocalTmpDir() {
 	return str + "\\Qalculate";
 #	endif
 #else
-	const char *homedir;
-	if((homedir = getenv("XDG_CACHE_HOME")) == NULL) {
-		return string(getpwuid(getuid())->pw_dir) + "/.cache/qalculate";
-	}
-	return string(homedir) + "/qalculate";
+	return string(getenv("SNAP_USER_DATA"));
 #endif
 }
 
@@ -551,7 +539,10 @@ bool move_file(const char *from_file, const char *to_file) {
 
 string getPackageDataDir() {
 #ifndef WIN32
-	return PACKAGE_DATA_DIR;
+	const char *snap = getenv("SNAP");
+	string datadir(snap);
+	datadir += "/share";
+ 	return datadir;
 #else
 	char exepath[MAX_PATH];
 	GetModuleFileName(NULL, exepath, MAX_PATH);
@@ -570,7 +561,7 @@ string getPackageDataDir() {
 
 string getGlobalDefinitionsDir() {
 #ifndef WIN32
-	return string(PACKAGE_DATA_DIR) + "/qalculate";
+	return getPackageDataDir() + "/qalculate";
 #else
 	char exepath[MAX_PATH];
 	GetModuleFileName(NULL, exepath, MAX_PATH);
@@ -600,7 +591,7 @@ string getGlobalDefinitionsDir() {
 
 string getPackageLocaleDir() {
 #ifndef WIN32
-	return PACKAGE_LOCALE_DIR;
+	return getPackageDataDir() + "/locale";
 #else
 	char exepath[MAX_PATH];
 	GetModuleFileName(NULL, exepath, MAX_PATH);
