diff --git a/libqalculate/Calculator-plot.cc b/libqalculate/Calculator-plot.cc
index 4081e4c..8eb083a 100644
--- a/libqalculate/Calculator-plot.cc
+++ b/libqalculate/Calculator-plot.cc
@@ -184,6 +184,7 @@ bool Calculator::plotVectors(PlotParameters *param, const vector<MathStructure>
 			commandline_extra += " -mono";
 		}
 		plot += "set terminal pop\n";
+		plot += "set terminal x11 font \"mbfont:sans\"\n";
 	} else {
 		persistent = true;
 		if(param->filetype == PLOT_FILETYPE_AUTO) {
